src/librustc_typeck/check/op.rs

```diff
@@ -289,6 +289,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                 // This has nothing here because it means we did string
                                 // concatenation (e.g. "Hello " + "World!"). This means
                                 // we don't want the note in the else clause to be emitted
+                            } else if let ty::TyParam(_) = lhs_ty.sty {
+                                // FIXME: point to span of param
```

Comment:
I have not been able to figure out how to get the span of `T` in the generic argument list in order to create a suggestion. The `FnCtxt` doesn't seem to store the generics or the `DefId` or the `BodyId` of the function it's currently checking.

Reply:
Nope. Presumably we could thread this information down.

Reply:
OK. I'll do that in a separate PR. This one already improves the message imo

Reply:
If we're going to solve this though, I'd really like to find a way to remove the `Name` from the `ty::ParamIdx` -- which presumably would require storing the set of in-scope generics into TLS somewhere (though that would possibly only be used by the `Debug` and `Display` impls).